### ISSUE TITLE
[bug] Wrong typings in AEMContainerComponent and AEMAllowedComponentsContainerComponent

### ISSUE THREAD

**Issue by cwuerzlhuber** [adobe/aem-angular-editable-components#41]
**Describe the bug**
In newer versions of the package, there are some typing issues on some components, which are making problems when upgrading from 1.0.3 to 1.2.0.

Some methods of components are wrongly type and breaking the build when upgrading to newer version.
Two examples broke our build (we have built our own containerComponent which inherits from AEMContainerComponent  with additional options):

E.g. AEMContainerComponent
```
getItem(itemKey: string): string {
  return this.cqItems && this.cqItems[itemKey];
}
```
return type is string but should be Model, since cqItems typing is `{ [key: string]: Model };`

Another example 
AEMAllowedComponentsContainerComponent
```
getAllowedComponents(): string[] {
  return this.allowedComponents && this.allowedComponents.components || [];
}
```
return type is string[] although allowedComponents.components is of type `AllowedComponent[]`

**Package version**
1.2.0

**To Reproduce**
1. Extend component from AEMContainerComponent
2. use getItem() method from super class
3. result will be typed with string, but is actually holding the cqItem

**Expected behavior**
Typing is correct and is not breaking the build when inheriting from components

**Additional context**
Generally there could be more issues related to this, but those two examples broke our build when trying to update to 1.2.0

**Comment by pfauchere** [comment]
Internally tracked by CQ-4319636

**Comment by niekraaijmakers** [comment]
Pr created for review.

**Comment by sharanyavinod** [comment]
PR merged and included in release v1.2.1
